crates/goose-server/src/routes/reply.rs

```diff
@@ -275,6 +273,13 @@ async fn handler(
 
     // Get a lock on the shared agent
     let agent = state.agent.clone();
+        
+    // Lock the agent to get a mutable reference so we can add the system
+    {
```

Comment:
I'm not familiar with this syntax but seems fine

Reply:
assuming you checked it works, its a bit confusing with the blocks and agents, can the initial agent be mutable?

Reply:
going to test it with the other branch selecting directories